devtools/etdump/data_sink.h

```diff
@@ -0,0 +1,28 @@
+// (c) Meta Platforms, Inc. and affiliates. Confidential and proprietary.
+
+#pragma once
+
+#include <executorch/devtools/etdump/data_sink_base.h>
+#include <executorch/runtime/core/exec_aten/exec_aten.h>
+#include <executorch/runtime/core/span.h>
+
+namespace executorch {
+namespace etdump {
+
+class DataSink : public DataSinkBase {
+ public:
+  explicit DataSink(::executorch::runtime::Span<uint8_t> buffer)
+      : debug_buffer_(buffer), offset_(0) {}
+
+  size_t write_tensor(const executorch::aten::Tensor& tensor) override;
+  size_t get_storage_size() const override;
+  size_t get_used_bytes() const override;
+
+ private:
```

Comment:
Please make sure that this class isn't copyable: right now it might be because its fields are trivially-copyable. But if someone creates two copies and writes to them, their offsets will be out of sync, violating the expected behavior by potentially corrupting the underlying data.